vdr/vdr.go

```diff
@@ -162,7 +162,17 @@ func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *t
 		return err
 	}
 
-	key, err := r.keyStore.Resolve(controllers[0].CapabilityInvocation[0].ID.String())
+	var key crypto.Key
+outer:
+	for _, c := range controllers {
```

Comment:
I checked what `ResolveControllers` actually does and this function's (`Update()`) behaviour was incorrect and still is incorrect when you read RFC006, which states:

> - a key referenced from the `capabilityInvocation` section of the latest DID document version.
- a key referenced from the `capabilityInvocation` section of a controller. One of the entries in the `controller` field MAY refer to a different DID Document.
  The `capabilityInvocation` keys of the latest version of that DID Document can be used as authorized key.

But `ResolveControllers()` returns _either_ the document itself as controller when is has no controller, but has capabilityInvocation keys, _or_ the controllers, not being the document itself. So according to RFC006 when the document has controllers (not including the subject DID itself) and a capabiltyInvocation key it can be used to upadte the DID document, but this `Update()` function won't use it (due to the implementation of `ResolveControllers`).

Reply:
Then the RFC is dodgy

Reply:
see nuts-foundation/nuts-specification#72